internal/cmd/common.go

```diff
@@ -0,0 +1,104 @@
+ * the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0'
+ */
+
+package cmd
+
+import (
+	"github.com/edgexfoundry/edgex-cli/internal/config"
+	"github.com/edgexfoundry/edgex-cli/internal/format"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/common"
+	"github.com/spf13/cobra"
+)
+
+var verbose, metadata, data, command, notification, scheduler, json bool
+
+func getSelectedService() string {
+	if metadata {
+		return common.CoreMetaDataServiceKey
+	} else if data {
+		return common.CoreDataServiceKey
+	} else if command {
+		return common.CoreCommandServiceKey
+	} else if notification {
+		return common.SupportNotificationsServiceKey
+	} else if scheduler {
+		return common.SupportSchedulerServiceKey
+	} else {
+		return ""
+	}
+}
+
+// commonServiceEndpointImplementation is used internally by config/metrics/ping/version
+// to implement the shared functionality of those commands
+func commonServiceEndpointImplementation(cmd *cobra.Command, api string, name string) error {
+	serviceName := getSelectedService()
+
+	if verbose {
+		if serviceName != "" {
+			s := config.GetCoreService(serviceName)
+			_, jsonValue, url, err := format.GetServiceEndpointValueAndJSON(s, api, name)
+			if err != nil {
+				return err
+			} else {
+				cmd.Printf("%s: %s: %s\n", serviceName, url, jsonValue)
+			}
+		} else {
+			format.PrintServices(config.GetCoreServices(), api, true, "")
+		}
+	} else if json {
+		// In the case of JSON output, we always print the result from a specified service,
+		// core-metadata by default
+		if serviceName == "" {
+			serviceName = common.CoreMetaDataServiceKey
+		}
+		s := config.GetCoreService(serviceName)
+		_, jsonStr, _, err := format.GetServiceEndpointValueAndJSON(s, api, "")
+		if err != nil {
+			return err
+		} else {
+			cmd.Println(jsonStr)
+		}
+	} else {
+		if serviceName != "" {
+			s := config.GetCoreService(serviceName)
+			value, _, _, err := format.GetServiceEndpointValueAndJSON(s, api, name)
+			if err != nil {
+				return err
+			} else {
+				cmd.Printf("%s: %s\n", serviceName, value)
+			}
+		} else {
+			format.PrintServices(config.GetCoreServices(), api, false, name)
+		}
+	}
+	return nil
+}
+
+func addFormatFlags(cmd *cobra.Command) {
+	cmd.Flags().BoolVarP(&json, "json", "j", false, "get JSON reply")
+	cmd.Flags().BoolVarP(&verbose, "debug", "d", false, "show debug output")
+
+}
+
+func addStandardFlags(cmd *cobra.Command) {
+	addFormatFlags(cmd)
+	cmd.Flags().BoolVarP(&data, "data", "", false, "use core-data service endpoint")
+	cmd.Flags().BoolVarP(&command, "command", "c", false, "use core-command service endpoint")
+	cmd.Flags().BoolVarP(&metadata, "metadata", "m", false, "use core-metadata service endpoint")
+	cmd.Flags().BoolVarP(&scheduler, "scheduler", "s", false, "use support-scheduler service endpoint")
+	cmd.Flags().BoolVarP(&notification, "notification", "n", false, "use support-notification service endpoint")
```

Comment:
```suggestion
	cmd.Flags().BoolVarP(&notifications, "notifications", "n", false, "use support-notifications service endpoint")
```
since the service's actual name is support-notifications

Reply:
thanks. fixed.